scripts/helpers/openai-compatible-agent.py

```
@@ -30,25 +53,47 @@ def tool_exec(cwd: Path, name: str, args: dict) -> str:
         if name == "run_command":
             cmd = str(args.get("command", ""))
             if len(cmd) > 600: return "ERROR: command too long"
-            r = subprocess.run(cmd, cwd=str(cwd), shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, timeout=20)
+            timeout = env_float("OPENAI_COMPAT_COMMAND_TIMEOUT", 20.0)
+            r = subprocess.run(cmd, cwd=str(cwd), shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, timeout=timeout)
```

Ruff (0.15.15)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)